pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -672,6 +673,42 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, ru
 	return
 }
 
+func (m *kubeGenericRuntimeManager) preStopSidecars(pod *v1.Pod, containers []*kubecontainer.Container) {
+	wg := sync.WaitGroup{}
+	wg.Add(len(containers))
+
+	for _, container := range containers {
+		go func(container *kubecontainer.Container) {
+			defer wg.Done()
+			gracePeriod := int64(minimumGracePeriodInSeconds)
```

Comment:
NIT - grace period is common to all containers so you don't have to execute them in every go routine